cli/src/registry/publish.ts

```diff
@@ -0,0 +1,227 @@
+  console.log(`Creating release artifacts for project: upgrader...`);
+  execSync(`BUILD_MODE='${network}' npx nx run upgrader:create-artifacts`);
+
+  switch (app) {
+    case Application.Station: {
+      const stationVersion = cargoProjectVersion(Application.Station);
+      console.log(`Creating release artifacts for project: station...`);
+      execSync(`BUILD_MODE='${network}' npx nx run station:create-artifacts`);
+
+      return {
+        name: applicationToRegistryEntryMap.station,
+        description:
+          'A secure and multichain asset management platform for teams and institutions.',
+        tags: ['latest'],
+        categories: ['finance', 'iam', 'multichain', 'asset-management'],
+        dependencies: [
+          {
+            name: applicationToRegistryEntryMap.upgrader,
+            version: upgraderVersion,
+          },
+        ],
+        module: await readFileIntoUint8Array(join(artifactsRootPath, 'station/station.wasm.gz')),
+        version: stationVersion,
+        metadata: {
+          url: 'https://github.com/dfinity/orbit',
+        },
+      };
+    }
+    case Application.Upgrader:
+      return {
+        name: applicationToRegistryEntryMap.upgrader,
+        description: 'Securely handles upgrades and disaster recovery for the Orbit Station.',
+        tags: ['latest'],
+        categories: ['finance', 'iam', 'multichain', 'asset-management'],
+        dependencies: [],
+        module: await readFileIntoUint8Array(join(artifactsRootPath, 'upgrader/upgrader.wasm.gz')),
+        version: upgraderVersion,
+        metadata: {
+          url: 'https://github.com/dfinity/orbit',
+        },
+      };
+  }
+};
+
+const command = createCommand('publish').description(
+  'Handles the publishing of the Orbit canisters to the control-panel registry.',
+);
+
+command
+  .option(
+    '-n, --network <TYPE>',
+    'The network to use for the registry operations. Defaults to `local`.',
+    'local',
+  )
+  .option(
+    '-i, --identity <TYPE>',
+    'The identity to use for the registry operations. Defaults to `default`.',
+    'default',
+  )
+  .requiredOption(
+    '-a, --app <VALUE>',
+    'The application to add to the registry. Must be either `station` or `upgrader`.',
+    parseRegistryApplication,
+  );
+
+// Saves the argument in a temporary file and returns the path to the file.
+const saveArgumentInTempFile = async (argument: string): Promise<string> => {
+  const tempFilePath = join(tmpdir(), 'orbit-cli-argument-' + Math.random().toString(36).slice(2));
+
+  await writeFile(tempFilePath, argument, {
+    encoding: 'utf-8',
+  });
+
+  return tempFilePath;
+};
+
+command.action(async options => {
+  const ping: { replica_health_status?: string } = JSON.parse(
+    await execAsync(`dfx ping '${options.network}'`),
+  );
+
+  if (ping.replica_health_status?.toLowerCase() !== 'healthy') {
+    throw new Error('The replica is not healthy.');
+  }
```

Comment:
Why not use `assertReplicaIsHealthy` here?

Reply:
Missed that one, just added now